dotnet/src/Connectors/Connectors.UnitTests/Memory/MemoryStorageTestBase.cs

```diff
@@ -0,0 +1,696 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.AI.Embeddings;
+using Microsoft.SemanticKernel.Memory;
+using Xunit;
+
+namespace SemanticKernel.Connectors.UnitTests.Memory;
+public abstract class MemoryStorageTestBase
+{
+    protected virtual string CreateRandomCollectionName()
+    {
+        return $"Collection_{Guid.NewGuid()}";
+    }
+
+    /// <summary>
+    /// Create <see cref="IMemoryStore"/> instance for unit tests
+    /// </summary>
+    /// <remarks>
+    /// you can override <see cref="CloseStoreAsync"/> in case you need extra clean up control
+    /// </remarks>
+    /// <returns><see cref="IMemoryStore"/> The instance for tests</returns>
+    protected abstract Task<IMemoryStore> CreateStoreAsync();
+
+    /// <summary>
+    /// Overide to add cleanup code for your <see cref="IMemoryStore"/> instance after unit test
+    /// </summary>
+    /// <param name="memoryStore"></param>
+    /// <returns></returns>
+    protected virtual Task CloseStoreAsync(IMemoryStore memoryStore)
+    {
+        return Task.CompletedTask;
+    }
+
+    protected virtual async Task WithStorageAsync(Func<IMemoryStore, Task> factAsync)
+    {
+        var store = await this.CreateStoreAsync();
+        await factAsync(store);
+        await this.CloseStoreAsync(store);
+    }
```

Comment:
Is this method actually needed? I think it complicates actual unit test, since unit test body will always be nested block of `.WithStorageAsync` method. Also, in xUnit it is possible to use class constructor for resource creation and implement `IDisposable` for resource clean-up. Which means that for storage creation/clean-up we can use dotnet native approach.

Reply:
Hi @dmytrostruk :
The reason I use the method is to avoid use a protected field in the class. to init with class constructor, I have to define an internal field like "protected IMemoryStore MyStore;" and init it in the constructor in the inhertited instance.